consensus.go

```diff
@@ -46,3 +47,23 @@ func (c *Consensus) CommitState(state consensus.State) (consensus.State, error)
 	}
 	return c.actor.SetState(state)
 }
+
+// Subscribe returns a channel on which every new state is sent.
+func (c *Consensus) Subscribe() <-chan consensus.State {
+	c.chMux.Lock()
+	defer c.chMux.Unlock()
+	if c.subscriberCh == nil {
```

Comment:
This feels like something we should just create in the constructor. throwing a lock around its optional existence makes things a bit more complicated.